src/webhooks/sentry-options/sentry-options.ts

```diff
@@ -205,23 +121,84 @@ export async function messageSlack(message: SentryOptionsResponse) {
 }
 
 async function sendMessage(blocks) {
-  for (const splitBlock of blocks) {
-    try {
-      await bolt.client.chat.postMessage({
-        channel: FEED_OPTIONS_AUTOMATOR_CHANNEL_ID,
-        blocks: splitBlock,
-        text: '',
-        unfurl_links: false,
-      });
-    } catch (err) {
-      Sentry.captureException(err);
-    }
+  try {
+    await bolt.client.chat.postMessage({
+      channel: FEED_OPTIONS_AUTOMATOR_CHANNEL_ID,
+      blocks: blocks,
+      text: '',
+      unfurl_links: false,
+    });
+  } catch (err) {
+    Sentry.setContext('block:', { blocks });
+    Sentry.captureException(err);
+  }
+}
+
+function generateBlock(option_type: string, options: any[]): KnownBlock[] {
+  /**
+   * This function generates a list of KnownBlocks, a type of SlackBlock.
+   * If the given option_type does not fit into the formatterMap, report it as a sentry error.
+   * This function also builds blocks around the options in batches of MAX_BLOCK_SIZE. Each
+   * sectionBlock has a block limit.
+   *
+   */
+
+  if (options.length === 0) {
+    return [];
+  }
+
+  const formatterMap: { [key: string]: OptionFormatter } = {
+    drifted: (option) =>
+      `\`${option.option_name}\` drifted. Value on db: \`${option.option_value}\``,
+    updated: (option) =>
+      `Updated \`${option.option_name}\` with db value \`${option.db_value}\` to value \`${option.value}\``,
+    set: (option) =>
+      `Set \`${option.option_name}\` to value \`${option.option_value}\``,
+    unset: (option) => `Unset \`${option}\``,
+    not_writable: (option) =>
+      `Failed to update \`${option.option_name}\` \nreason: \`${option.error_msg}\``,
+    unregistered: (option) => `Option \`${option}\` is not registered!`,
+    invalid_type: (option) =>
+      `Option \`${option.option_name}\` got type \`${option.got_type}\`, but expected type \`${option.expected_type}\`.`,
+  };
+
+  const blocks: KnownBlock[] = [];
+  blocks.push(slackblocks.divider());
+
+  if (formatterMap[option_type]) {
+    blocks.push(
+      ...createOptionBlocks(options, option_type, formatterMap[option_type])
+    );
+    return blocks;
+  } else {
+    Sentry.captureException(`unsupported option type: ${option_type}`);
+    return [];
   }
 }
-function splitMessage(block: KnownBlock[]): KnownBlock[][] {
-  const splitBlock: KnownBlock[][] = [];
-  for (let i = 0; i < block.length; i += 10) {
-    splitBlock.push(block.slice(i, i + 10));
+
+function createOptionBlocks(
+  options: any[],
+  option_type: string,
+  formatter: OptionFormatter
+): KnownBlock[] {
+  const block: KnownBlock[] = [];
+  const header = `*${option_type.charAt(0).toUpperCase()}${option_type.slice(
```

Comment:
Won't this also include the `_` in the option types?

Reply:
yeah, it will. I think that's fine though, it's still very readable, parsing it out for some of the cases doesn't seem necessary to me.